File: gui_view/src/main/scala/com/goexp/galgame/gui/view/brand/InfoController.java
package com.goexp.galgame.gui.view.brand;

import com.goexp.galgame.gui.model.Brand;
import com.goexp.galgame.gui.model.Game;
import com.goexp.galgame.gui.task.TaskService;
import com.goexp.galgame.gui.task.game.search.ByBrand;
import com.goexp.galgame.gui.view.DefaultController;
import com.goexp.galgame.gui.view.game.listview.DataViewController;
import javafx.beans.value.ChangeListener;
import javafx.collections.ObservableList;
import javafx.collections.transformation.FilteredList;
import javafx.concurrent.Service;
import javafx.fxml.FXML;

public class InfoController extends DefaultController {

    /**
     * UI Com
     */

    @FXML
    public TitlePartController titleController;

    @FXML
    public DataViewController dataViewController;

    private FilteredList<Game> filteredGames;


    /***
     * Biz
     */

    private Brand brand = new Brand();

    /**
     * Service
     */


    private Service<ObservableList<Game>> gameByBrand = new TaskService<>(() -> new ByBrand(brand.id()));


    /**
     * Event
     */


    protected void initialize() {

        gameByBrand.valueProperty().addListener((observable1, oldValue1, newValue1) -> {

            if (newValue1 != null) {
                load(newValue1);
            }

        });

        dataViewController.progessloading.visibleProperty().bind(gameByBrand.runningProperty());
        dataViewController.reloadProperty.addListener((observable, oldValue, newValue) -> {
            if (newValue != null && newValue) {
                gameByBrand.restart();
            }
        });

    }


    private void load(ObservableList<Game> games) {
        filteredGames = new FilteredList<>(games);
        dataViewController.load(filteredGames);
    }

    public void load(Brand brand) {

        logger.info("<Brand> {}", brand);
        this.brand = brand;

        titleController.init(brand);
        gameByBrand.restart();
    }

}

File: gui_view/src/main/scala/com/goexp/galgame/gui/view/brand/InfoController.scala
package com.goexp.galgame.gui.view.brand

import java.util.Objects

import com.goexp.galgame.gui.model.{Brand, Game}
import com.goexp.galgame.gui.task.TaskService
import com.goexp.galgame.gui.task.game.search.ByBrand
import com.goexp.galgame.gui.view.DefaultController
import com.goexp.galgame.gui.view.game.listview.DataViewController
import javafx.collections.ObservableList
import javafx.collections.transformation.FilteredList
import javafx.fxml.FXML

class InfoController extends DefaultController {

  @FXML var titleController: TitlePartController = _
  @FXML var dataViewController: DataViewController = _

  private var brand = new Brand

  final private val gameByBrand = TaskService(new ByBrand(brand.id))

  override protected def initialize() = {
    gameByBrand.valueProperty.addListener((_, _, newValue) => {
      if (newValue != null) load(newValue)
    })

    dataViewController.progessloading.visibleProperty.bind(gameByBrand.runningProperty)

    dataViewController.reloadProperty.addListener((_, _, newValue) => {
      if (newValue) gameByBrand.restart()

    })
  }

  private def load(games: ObservableList[Game]) = {
    val filteredGames = new FilteredList[Game](games)
    dataViewController.load(filteredGames)
  }

  def load(brand: Brand) = {
    Objects.requireNonNull(brand)

    logger.info(s"Brand[${brand.id}] ${brand.name} state:<${brand.isLike}>")

    this.brand = brand
    titleController.init(brand)
    gameByBrand.restart()
  }
}